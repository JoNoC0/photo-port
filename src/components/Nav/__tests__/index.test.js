// __tests__/Nav.test.js with hard coded categories
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe('Nav component renders', () => {
  it('renders', () => {
    render(<Nav />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Nav />);
<<<<<<< HEAD
<<<<<<< HEAD

=======
    
>>>>>>> feature/react-tests
=======
    
>>>>>>> feature/contact
    expect(asFragment()).toMatchSnapshot();
  });
})

describe('emoji is visible', () => {
  it('inserts emoji into the h2', () => {
<<<<<<< HEAD
<<<<<<< HEAD
    const { getByLabelText } = render(<Nav />);

    expect(getByLabelText('camera')).toHaveTextContent('ðŸ“¸');
  });
})
=======
=======
>>>>>>> feature/contact
  const { getByLabelText } = render(<Nav />);

  expect(getByLabelText('camera')).toHaveTextContent('ðŸ“¸');
  });
})  
<<<<<<< HEAD
>>>>>>> feature/react-tests
=======
>>>>>>> feature/contact

describe('links are visible', () => {
  it('inserts text into the links', () => {
    const { getByTestId } = render(<Nav />);

    expect(getByTestId('link')).toHaveTextContent('Oh Snap!');
    expect(getByTestId('about')).toHaveTextContent('About me');
  });

})