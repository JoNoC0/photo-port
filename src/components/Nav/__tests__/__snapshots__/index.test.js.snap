// Jest Snapshot v1, https://goo.gl/fbAQLP
<<<<<<< HEAD
<<<<<<< HEAD
exports[`Nav component matches snapshot 1`] = `
=======

exports[`Nav component renders matches snapshot 1`] = `
>>>>>>> feature/contact
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h2>
      <a
        data-testid="link"
<<<<<<< HEAD
=======

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
>>>>>>> feature/react-tests
=======
>>>>>>> feature/contact
        href="/"
      >
        <span
          aria-label="camera"
          role="img"
        >
           ðŸ“¸
        </span>
         Oh Snap!
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
<<<<<<< HEAD
<<<<<<< HEAD
            data-testid="about"
=======
>>>>>>> feature/react-tests
=======
            data-testid="about"
>>>>>>> feature/contact
            href="#about"
          >
            About me
          </a>
        </li>
<<<<<<< HEAD
<<<<<<< HEAD
        <li
          class="mx-2"
        >
=======
        <li>
>>>>>>> feature/react-tests
=======
        <li
          class="mx-2 undefined"
        >
>>>>>>> feature/contact
          <span>
            Contact
          </span>
        </li>
<<<<<<< HEAD
<<<<<<< HEAD
=======
        <li
          class="mx-1"
        >
          <span>
            commercial
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            portraits
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            food
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            landscape
          </span>
        </li>
>>>>>>> feature/react-tests
=======
>>>>>>> feature/contact
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
